File: 31Jan2024.java
import java.util.Stack;

class Solution {
    public int[] dailyTemperatures(int[] t) {
        Stack<Integer> stk = new Stack<>();
        int n = t.length;
        int[] ans = new int[n];
        for (int i = t.length - 1; i >= 0; i--) {
            while (stk.size() > 0 && t[stk.peek()] <= t[i])
                stk.pop();
            if (stk.size() == 0) {
                ans[i] = 0;
            } else {
                ans[i] = stk.peek() - i;
            }
            stk.push(i);
        }
        return ans;
    }
}

File: 31Jan2024.py
from ast import List


class Solution:
    def dailyTemperatures(self, t: List[int]) -> List[int]:
        stk = []
        n = len(t)
        ans = [0]*n
        for i in range(n-1,-1,-1) :
            while(len(stk)>0 and t[stk[-1]]<=t[i]) :
                stk.pop()
            if(len(stk)==0) :
                ans[i]=0
            else :
                ans[i]=stk[-1]-i
            stk.append(i)
        return ans